### PR TITLE
Silencing few build warning.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: lib/common/url.c [hunk 1]
```diff
@@ -423,7 +423,7 @@ int h2o_url_init_with_hostport(h2o_url_t *url, h2o_mem_pool_t *pool, const h2o_u
     } else {
         url->_port = port;
         char _port[sizeof(H2O_UINT16_LONGEST_STR)];
-        int port_len = sprintf(_port, "%" PRIu16, port);
+        int port_len = sprintf(_port, "%h" PRIu16, port);
```
Comment (kazuho) [inline lib/common/url.c]: What is the warning you are seeing?

`port` is `uint16_t`; and IIUC the type specifier _is_ `"%" PRIu16`.

Comment (devnexen) [reply lib/common/url.c]: Was unable to reproduce the warning I got a while ago.

File: lib/handler/compress/brotli.cc [hunk 2]
```diff
@@ -72,8 +72,7 @@ void brotli_context::_emit(bool is_last, bool force_flush)
     uint8_t *output;
     size_t out_size;
     bool ret = brotli_->WriteBrotliData(is_last, force_flush, &out_size, &output);
-    assert(ret);
-    if (out_size != 0)
+    if (ret && out_size != 0)
```
Comment (kazuho) [inline lib/handler/compress/brotli.cc]: Does changing this to the below solve the issue? 
```
assert(ret);
(void)ret;
if (out_size != 0)
```

The intent here is to retain the behavior as-is (we want to check the behavior in debug builds but not in release builds).

Comment (devnexen) [reply lib/handler/compress/brotli.cc]: Sure it will solve :-)

Comment (deweerdt) [reply lib/handler/compress/brotli.cc]: I've seen this:
```(86 of 782): warning: ‘%u’ directive writing between 1 and 10 bytes into a region of size 6 [-Wformat-overflow=]                                                                                                                                                                                            426,39        99%
```

**Pull request closed without merging**